subsys/bluetooth/host/att.c

```diff
@@ -710,25 +642,24 @@ static struct net_buf *bt_att_chan_create_pdu(struct bt_att_chan *chan, uint8_t
 		timeout = K_FOREVER;
 	}
 
-	buf = bt_l2cap_create_pdu_timeout(NULL, 0, timeout);
+	/* This will reserve headspace for lower layers */
+	buf = bt_l2cap_create_pdu_timeout(&att_pool, 0, timeout);
 	if (!buf) {
 		LOG_ERR("Unable to allocate buffer for op 0x%02x", op);
 		return NULL;
 	}
 
-	data = tx_meta_data_alloc(timeout);
-	if (!data) {
-		LOG_WRN("Unable to allocate ATT TX meta");
-		net_buf_unref(buf);
-		return NULL;
-	}
+	data = bt_att_get_tx_meta_data(buf);
+	/* If we got a buf from `att_pool`, then the metadata slot at its index
+	 * is officially ours to use.
+	 */
+	memset(data, 0, sizeof(*data));
```

Comment:
If we memset to 0 on destroy, we can skip the memset here

Reply:
see other comment.